File: ui/lib/core/Glyph.ts
import { Contour } from "./Contour";

export class Glyph {
  #name: string;
  #contours: Contour[];

  constructor(name: string, contours: Contour[]) {
    this.#name = name;
    this.#contours = contours;
  }

  get name(): string {
    return this.#name;
  }

  get contours(): Contour[] {
    return this.#contours;
  }
}

File: crates/shift-font/src/glyph.rs
use serde::Serialize;
use ts_rs::TS;

use crate::contour::Contour;

#[derive(Clone, Serialize, TS)]
#[ts(export)]
pub struct Glyph {
    name: String,
    unicode: u32,
    contours: Vec<Contour>,
    x_advance: f64,
}

impl Glyph {
    pub fn new(name: String, unicode: u32, contours: Vec<Contour>, x_advance: f64) -> Self {
        Self {
            name,
            unicode,
            contours,
            x_advance,
        }
    }

    pub fn get_name(&self) -> &str {
        &self.name
    }

    pub fn get_contours(&self) -> &Vec<Contour> {
        &self.contours
    }
}
